### ISSUE TITLE
[contracts] Compute all config rather than store in state

### ISSUE THREAD

**Issue by ponderingdemocritus** [BibliothecaDAO/eternum#601]
Currently throughout the app when we use a config value in models, we read the config then store that in the state of a model.

This hard codes the config value for all future interactions with this model.

We need to move this so all config values are computed on the fly, rather than hard coding when a model is created. This will reduce gas costs, and also allow the config to change within a session.

--------

This issue basically concerns the `CapacityConfig` and `SpeedConfig`. <br><br>
Regarding capacity configuration, you'll see that we hardcode the `Capacity` value of entities here
https://github.com/BibliothecaDAO/eternum/blob/cc2089f6632293d55cbbd5e11e470ca239f6f92e/contracts/src/systems/combat/contracts.cairo#L120

instead of doing that, we could store the entity type of in the `EntityMetadata` model
then whenever we need to get the capacity of the entity, we can get it by checking the current capacity config value of the entity type e.g
```
let entity_metadata: EntotyMetadata = get!(world, entity_id, EntityMetadata);
let capacity : CapacityConfig = CapacityConfigImpl::get(world, entity_metadata.entity_type);
```
then we use the capacity computed on the fly when determining the weight of entities
https://github.com/BibliothecaDAO/eternum/blob/cc2089f6632293d55cbbd5e11e470ca239f6f92e/contracts/src/models/weight.cairo#L10-L35
<br><br>

With regard to `SpeedConfig`, we will have to stop storing `ArrivalTime` and instead store the start time of a journey so that we can always calculate the arrival time on the fly by multiplying an entity's speed with the distance of journey
<br><br><br><br>

**Comment by EjembiEmmanuel** [comment]
@ponderingdemocritus can I work on this?

**Comment by credence0x** [comment]
@ponderingdemocritus i don't get the description. could you rephrase

**Comment by ponderingdemocritus** [comment]
When we use config values throughout the app, we often pull from the config model then hardcode the value into a model. We do this for `sec_per_km'. This means that if we change the config, all entities that existed in the past, now have the old config. 

If we move to computing the config on the fly whenever it is used, it means that we can change the config model, and every entity value changes.

**Comment by credence0x** [comment]
oh i see

**Comment by credence0x** [comment]
@EjembiEmmanuel i can walk you through it if you're willing to work on it asap

**Comment by Dprof-in-tech** [comment]
> @EjembiEmmanuel i can walk you through it if you're willing to work on it asap

I am willing to work on this though if he's not available

**Comment by EjembiEmmanuel** [comment]
@credence0x yes, you can I'm available to work on it.

**Comment by credence0x** [comment]
> > @EjembiEmmanuel i can walk you through it if you're willing to work on it asap
> 
> I am willing to work on this though if he's not available

seems he is. is there any other issue you'd like to take? @Dprof-in-tech

**Comment by credence0x** [comment]
> @credence0x yes, you can I'm available to work on it.

@EjembiEmmanuel pls text me on discord. @credence0x

**Comment by EjembiEmmanuel** [comment]
@credence0x I've sent you a friend request on Discord

**Comment by ponderingdemocritus** [comment]
yes - just do those

**Comment by edisontim** [comment]
Is this done or should I work on this?
